resources/migrations/056_update_migrations.yaml

```diff
@@ -169,7 +169,380 @@ databaseChangeLog:
+      preConditions:
+        - onFail: MARK_RAN
+        - tableExists:
+            tableName: data_app_definition
+        - not:
+            - indexExists:
+                tableName: data_app_definition
+                indexName: idx_data_app_definition_creator_id
+      changes:
+        - createIndex:
+            tableName: data_app_definition
+            indexName: idx_data_app_definition_creator_id
+            columns:
+              - column:
+                  name: creator_id
+      rollback: #no op, will be deleted with the table
+
+  - changeSet:
+      id: v56.2025-07-03T16:02:20
+      author: qnkhuat
+      comment: Add unique constraint on data_app_definition (app_id, revision_number)
+      preConditions:
+        - onFail: MARK_RAN
+        - tableExists:
+            tableName: data_app_definition
+        - not:
+            - uniqueConstraintExists:
+                tableName: data_app_definition
+                constraintName: uniq_idx_data_app_definition_app_id_revision_number
+      changes:
+        - addUniqueConstraint:
+            tableName: data_app_definition
+            constraintName: uniq_idx_data_app_definition_app_id_revision_number
+            columnNames: app_id, revision_number
+      rollback:
+        - dropUniqueConstraint:
+            tableName: data_app_definition
+            constraintName: uniq_idx_data_app_definition_app_id_revision_number
+
+  - changeSet:
+      id: v56.2025-07-03T16:03:00
+      author: qnkhuat
+      comment: Create data_app_release table for data apps functionality
+      preConditions:
+        - onFail: MARK_RAN
+        - not:
+            - tableExists:
+                tableName: data_app_release
+      changes:
+        - createTable:
+            tableName: data_app_release
+            remarks: App publishing records
+            columns:
+              - column:
+                  name: id
+                  remarks: Unique ID
+                  type: int
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: app_id
+                  remarks: Referenced app
+                  type: int
+                  constraints:
+                    nullable: false
+                    referencedTableName: data_app
+                    referencedColumnNames: id
+                    foreignKeyName: fk_data_app_release_app_id
+                    deleteCascade: true
+              - column:
+                  name: app_definition_id
+                  remarks: Referenced app definition
+                  type: int
+                  constraints:
+                    nullable: false
+                    referencedTableName: data_app_definition
+                    referencedColumnNames: id
+                    foreignKeyName: fk_data_app_release_app_definition_id
+                    deleteCascade: true
```

Comment:
Let's take this out, so we can't accidentally delete a definition for an active release.